src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -102,9 +111,22 @@ public Task RunAsync(ActionRunStage stage)
             actionRunner2.Action = cleanOutputsStep;
             actionRunner2.Stage = ActionRunStage.Main;
             actionRunner2.Condition = "always()";
-            ExecutionContext.RegisterNestedStep(actionRunner2, inputsData, location, Environment, true);
+            var cleanUpStep = ExecutionContext.RegisterNestedStep(actionRunner2, inputsData, location, Environment, true);
+            ExecutionContext.CompositeSteps.Add(cleanUpStep);
 
-            return Task.CompletedTask;
+            // Then run the Composite StepsRunner
+            try
+            {
+                await compositeStepsRunner.RunAsync(ExecutionContext);
+            }
+            catch (Exception ex)
+            {
+                // Composite StepRunner should never throw exception out.
+                Trace.Error($"Caught exception from composite steps {nameof(CompositeStepsRunner)}: {ex}");
+                ExecutionContext.Error(ex);
+                ExecutionContext.Result = TaskResult.Failed;
+            }
+            ExecutionContext.Result = TaskResult.Succeeded;
```

Comment:
you don't need to set `ExecutionContext.Result = TaskResult.Succeeded;`, the result defaults to succeed.